docs/src/pages/demos/snackbars/DirectionSnackbar.tsx

```diff
@@ -25,14 +25,10 @@ export interface State {
   Transition?: React.ComponentType<TransitionProps>;
 }
 
-class DirectionSnackbar extends React.Component<void, State> {
-  constructor() {
-    super();
-
-    this.state = {
-      open: false,
-    };
-  }
+class DirectionSnackbar extends React.Component<{}, State> {
+  state: State = {
```

Comment:
The generic argument in `React.Component` is required for correct `setState` type while the explicit annotation in the class field is required for correct types in `this.state`. Otherwise TypeScript will infer the apparent type (`{ open: boolean }`) which means `this.state.Transition` is access of a not existing key.